registry/auth/token/token.go

```diff
@@ -168,19 +168,23 @@ func (t *Token) Verify(verifyOpts VerifyOptions) error {
 	}
 
 	// Verify that the signing key is trusted.
-	signingKey, err := t.VerifySigningKey(verifyOpts)
+	signingKeys, err := t.VerifySigningKey(verifyOpts)
 	if err != nil {
 		log.Info(err)
 		return ErrInvalidToken
 	}
 
-	// Finally, verify the signature of the token using the key which signed it.
-	if err := signingKey.Verify(strings.NewReader(t.Raw), t.Header.SigningAlg, t.Signature); err != nil {
-		log.Infof("unable to verify token signature: %s", err)
-		return ErrInvalidToken
+	for _, key := range signingKeys {
+		// Finally, verify the signature of the token using the key which signed it.
+		err := key.Verify(strings.NewReader(t.Raw), t.Header.SigningAlg, t.Signature)
+		if err != nil {
+			log.Infof("unable to verify token signature: %s", err)
+		} else {
+			return nil
```

Comment:
I'd probably rewrite this part to use `continue` (not a fan of `else` clauses in Go) with early return
```Go
...
		if err != nil {
			log.Infof("unable to verify token signature: %s", err)
			continue
		} 
		return nil
...
```

Reply:
good shout, I think

```go
		err := key.Verify(strings.NewReader(t.Raw), t.Header.SigningAlg, t.Signature)
		if err == nil {
			return nil
		}
		log.Infof("unable to verify token signature: %s", err)
```
makes it a slightly bit more readable? I'm fine with either one

Reply:
Yeah, I was thinking about that, too, but then decided to suggest to do the Go error checking first as my brain is wired that way now -- every time I see `error` returned I immediately expect `err != nil` 😅

Reply:
I'll leave that to you to decide which one you prefer. I'm fine with either :)

Reply:
let's go with the `continue` one, I agree with automatically assuming it's `err != nil`